fix: Fixes table editable cell styles and simplifies it

pan-kot opened the pull request:
### Description

The PR refactors table body cell styles with the focus on editable cells. That is needed for the follow-up task to move editable cell style from the cell element to the cell content instead. This would allow having both row expand toggle and editable cell trigger in a single cell (in 1st column).

### How has this been tested?

* Screenshot tests
* Manual testing on the cell-permutations page

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -25,11 +25,10 @@ $edit-button-padding-right: calc(#{awsui.$space-xs} + #{awsui.$space-xxs});
 $cell-offset: calc(#{awsui.$space-m} + #{awsui.$space-xs});
 
-// Ensuring enough space for absolute-positioned focus outlines of focus-able cell content elements.
-$cell-negative-space-vertical: 2px;
```
This negative space was used to virtually enlarge the cell-content element. That is because the element has `overflow: hidden` but when there are interactive elements inside such a link or a button dropdown - the focus outline is being cut off.

The negative space is no longer needed because now the cell content size matches cell size. This way the outline can use the available cell space while the content is still truncated correctly.

pan-kot replied:
Update: the negative space is still needed at least for the compact mode.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -131,11 +127,18 @@ $cell-negative-space-vertical: 2px;
+      align-content: baseline;
+    }
+
+    &:not(.body-cell-wrap) {
```
The `.body-cell-wrap` only takes effect on the cell-content element so attaching the style to it directly.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -177,20 +180,10 @@ $cell-negative-space-vertical: 2px;
-    // That is to prevent focus outline on cell content from being cut out.
-    > .body-cell-content {
-      padding-inline-start: awsui.$border-divider-list-width;
-      margin-inline-start: calc(-1 * #{awsui.$border-divider-list-width});
```
This negative space shenanigans are no longer needed because there is plenty of space for the focus outline.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -380,19 +373,6 @@ $cell-negative-space-vertical: 2px;
       color: awsui.$color-text-button-normal-active;
     }
 
-    &-form {
```
For editing cells we used `overflow: visible` on cell content together with negative margins so that the editor can take more space inside a cell and its focus outline is not cut off. Instead, in the refactored versions we just use different padding for editing cells and there is no longer an issue with the outline.

pan-kot replied:
Update: the overflow visible is still needed for editable content that includes dropdowns. I will contribute a new screenshot test to capture that.

pan-kot commented on src/table/body-cell/td-element.tsx:
```diff
@@ -90,6 +90,7 @@ export const TableTdElement = React.forwardRef<HTMLTableCellElement, TableTdElem
       collapseButtonLabel,
       verticalAlign,
       resizableColumns,
+      resizableStyle,
```
The `resizableStyle` replaces `style` to make the cell API better predictable so that no unexpected attributes can be assigned from the outside.

pan-kot commented on src/table/styles.scss:
```diff
@@ -53,6 +53,7 @@
 
 .table {
   inline-size: 100%;
+  block-size: 100%;
```
This change does not affect the table itself - the size of the table is determined by the size of its content.

However, the table element needs the height set for cells height to take an effect, see discussion: https://stackoverflow.com/questions/3215553/make-a-div-fill-an-entire-table-cell.

georgylobko commented on src/table/body-cell/td-element.tsx:
```diff
@@ -101,6 +102,8 @@ export const TableTdElement = React.forwardRef<HTMLTableCellElement, TableTdElem
     const Element = isRowHeader ? 'th' : 'td';
     const isVisualRefresh = useVisualRefresh();
 
+    resizableStyle = resizableColumns ? {} : resizableStyle;
```
this condition looks counterintuitive. you apply `resizableStyle` only when `resizableColumns` is false. why is that?

pan-kot replied:
That is taken as is from the current impl: https://github.com/cloudscape-design/components/blob/main/src/table/internal.tsx#L627

When resizable columns is on, the styles are assigned directly to cell nodes in `src/table/column-widths-utils.ts` and I think the properties like min-width and max-width are ignored.

I think there are further opportunities for refactoring and maybe this condition can be just removed but I'd not include this into the current PR.

YueyingLu commented:
Found 2 issues with manual testing:
1. can not open select dropdown 

https://github.com/user-attachments/assets/6682b553-b18c-44bd-89eb-2d8266ff114a



2. the input position seems a bit off 
<img width="639" alt="Screenshot 2024-12-09 at 22 28 15" src="https://github.com/user-attachments/assets/801f9116-8e6a-46ec-bb5a-99648c1c9924">

pan-kot commented:
@georgylobko, @YueyingLu 

I added a new commit with the following changes:
1. Reverted changes to negative vertical space. That is because in classic+compact the focus outline can be still cut off if the button is rendered inside content and the rows only have text content. The vertical paddings we use for table cell are under 2px whilst the focus outline offset for buttons is 2px - hence we need to do a trick so that the focus outline can fit.
2. Reverted changes to visible overflow in editable cells. When a cell is editing the overflow is allowed so that components with dropdown like Select can be displayed correctly.

I will re-run the screenshot tests for these changes, too.
